daemon/openrazer_daemon/daemon.py

```diff
@@ -326,7 +326,7 @@ def write_persistence(self, persistence_file):
             if 'set_dpi_xy' in device.dbus.METHODS or 'set_dpi_xy_byte' in device.dbus.METHODS:
                 dpi_x = int(device.dbus.dpi[0])
                 dpi_y = int(device.dbus.dpi[1])
-                if dpi_x > 0 and dpi_y > 0:
+                if dpi_x > 0 and dpi_y > 0 or 'available_dpi' in device.dbus.METHODS and dpi_y == 0:
```

Comment:
I think this should also work?
```python
# Devices with 'available_dpi' only have X values, Y will be 0
if dpi_x > 0 and (dpi_y > 0 or 'available_dpi' in device.dbus.METHODS):
```

Reply:
That works if the check for `dpi_y` being equal to zero is not needed. Zero is the only valid `dpi_y` value for DPI X only device. Without that check a negative value could be saved to the persistence file. I'm not sure how the value could end up negative but we don't have to find out if we keep the check. The additional check has no impact on performance.

Reply:
Without checking Python docs how an `and` - `or` chain gets evaluated I'd like to see some brackets in your solution to make it more obvious which `and` & `or` groups belong together. And add a comment above also to explain what this line should do, and the specifics about devices with `available_dpi`